crates/biome_js_analyze/src/lint/nursery/use_collapsed_if.rs

```diff
@@ -0,0 +1,198 @@
+            // `JsIfStatement`, the child `if` statement should be merged.
+            AnyJsStatement::JsBlockStatement(parent_block_statement) => {
+                let statements = parent_block_statement.statements();
+                if statements.len() != 1 {
+                    return None;
+                }
+
+                let AnyJsStatement::JsIfStatement(child_if_statement) = statements.first()? else {
+                    return None;
+                };
+
+                Some(child_if_statement)
+            }
+            // If `consequent` is a `JsIfStatement` without any block, it should be merged.
+            AnyJsStatement::JsIfStatement(child_if_statement) => Some(child_if_statement),
+            _ => None,
+        }?;
+
+        // It cannot be merged if the child `if` statement has any else clause(s).
+        if child_if_statement.else_clause().is_some() {
+            return None;
+        }
+
+        Some(RuleState {
+            parent_if_statement: if_stmt.clone(),
+            child_if_statement,
+        })
+    }
+
+    fn diagnostic(_ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            state.child_if_statement.syntax().text_range(),
+            markup! {
+                "This "<Emphasis>"if"</Emphasis>" statement can be collapsed into another "<Emphasis>"if"</Emphasis>" statement."
+            },
+        ))
+    }
+
+    fn action(ctx: &RuleContext<Self>, state: &Self::State) -> Option<JsRuleAction> {
+        let RuleState {
+            parent_if_statement,
+            child_if_statement,
+        } = state;
+
+        let parent_consequent = parent_if_statement.consequent().ok()?;
+        let parent_test = parent_if_statement.test().ok()?;
+        let child_consequent = child_if_statement.consequent().ok()?;
+        let child_test = child_if_statement.test().ok()?;
+
+        let parent_has_comments = match &parent_consequent {
+            AnyJsStatement::JsBlockStatement(block_stmt) => {
+                block_stmt.l_curly_token().ok()?.has_trailing_comments()
+                    || block_stmt.r_curly_token().ok()?.has_leading_comments()
+            }
+            _ => false,
+        };
+
+        let has_comments = parent_has_comments
+            || child_if_statement.syntax().has_comments_direct()
+            || child_if_statement
+                .r_paren_token()
+                .ok()?
+                .has_trailing_comments();
+        if has_comments {
+            return None;
+        }
+
+        let mut expr =
+            js_logical_expression(parent_test.clone(), token(T![&&]), child_test.clone());
+
+        // Parenthesize arms of the `&&` expression if needed
+        let left = expr.left().unwrap();
+        if left.needs_parentheses() {
+            expr = expr.with_left(parenthesized(left).into());
+        }
+
+        let right = expr.right().unwrap();
```

Comment:
by the way, we try to avoid `unwrap` because it is the source of many panics.
Even if here we know it is not None, we could just write:
```suggestion
        let left = expr.left().ok()?;
        if left.needs_parentheses() {
            expr = expr.with_left(parenthesized(left).into());
        }

        let right = expr.right().ok()?;
```

Reply:
Fixed at [68c6810](https://github.com/biomejs/biome/pull/4179/commits/68c6810c1db38766112db18d90f9c8dba9301fc0)!